utils/containers/hash/hash_map.h.tpl

```diff
@@ -12,6 +12,7 @@
 
 #include "common/errors.h"
 #include "common/trinary/flex_trit.h"
+#include "consensus/model.h"
```

Comment:
I think we should find another way to achieve this. If we progressively include directly in the template we will end up with containers having a lot of unnecessary dependencies.
E.g. we could add an additional `include` parameter to the `hash_map_generate` rule.

Reply:
Had that in mind, though eventually those are only types

Reply:
will do

Reply:
note that this will also pull in a dependency on the final rule, also we should then no longer have the generated files live in `utils` but rather where the model/... is defined.